src/agent.rs

```diff
@@ -118,26 +119,29 @@ impl Agent {
         // Create the channels
         // TODO: make all components listen to shutdown signal
         let (shutdown_tx, _) = broadcast::channel(self.config.channel_capacities.shutdown);
-        let (primary_oracle_updates_tx, primary_oracle_updates_rx) =
-            mpsc::channel(self.config.channel_capacities.primary_oracle_updates);
-        let (secondary_oracle_updates_tx, secondary_oracle_updates_rx) =
-            mpsc::channel(self.config.channel_capacities.secondary_oracle_updates);
-        let (global_store_lookup_tx, global_store_lookup_rx) =
-            mpsc::channel(self.config.channel_capacities.global_store_lookup);
         let (local_store_tx, local_store_rx) =
             mpsc::channel(self.config.channel_capacities.local_store);
         let (primary_keypair_loader_tx, primary_keypair_loader_rx) = mpsc::channel(10);
         let (secondary_keypair_loader_tx, secondary_keypair_loader_rx) = mpsc::channel(10);
 
+        let mut registry = PROMETHEUS_REGISTRY.lock().await;
```

Comment:
This feels out of place. If we are just passing it to the adapter then it's passed down to global store, can't we initialize it in the global store?

Reply:
Yeah this is temporary. Before this change the static global one was used but my changes caused initialization code to not be async (good) but therefore no longer able to take the lock in those places (bad). This is also used in other places so as I refactor the other parts of the code, they will also need the metrics. So lifting it up to the top so I can pass it through temporarily is what I opted for.

Later, I will do what we do in hermes, and have a `Metrics` API that does the initialization and the other APIs will just use it exactly as you suggest.